client/components/utils/add-comment-plugin.js

```diff
@@ -78,22 +78,20 @@ export default function addCommentPlugin(vueInstance) {
 }
 
 export function addHighlight(keys, schema, state, from, to, dispatch) {
-  const {doc, tr} = state;
-  let attrs = null;
-  if (!doc.rangeHasMark(from, to, schema.marks.highlight)) {
-    attrs = {"highlight-keys": keys};
-    if (!attrs["highlight-keys"]) {
-      return false;
-    }
+  const {tr: transaction} = state;
+  const attrs = {"highlight-keys": keys};
+  if (!attrs["highlight-keys"]) {
+    return false;
   }
-  return dispatch(tr.addMark(from, to, schema.marks.highlight.create(attrs)));
+
+  return dispatch(transaction.addMark(from, to, schema.marks.highlight.create(attrs)));
 }
 
 export function removeHighlight(schema, state, from, to, dispatch) {
-  const {doc, tr} = state;
+  const {doc, tr: transaction} = state;
```

Comment:
I see. This is part of ProseMirror state. It is not a variable you named like this. Then it is probably better to keep it `tr` instead of renaming it, and just maybe add a comment that `"tr" is a ProseMirror transaction.`